scenario_player/services/utils/factories.py

```diff
@@ -45,3 +51,118 @@ def construct_flask_app(
         SP_PM.hook.register_blueprints(app=app)
 
     return app
+
+
+def daemonize(pidfile: pathlib.Path, *, stdin="/dev/null", stdout="/dev/null", stderr="/dev/null"):
+    """Daemonize the currently run script, using a double fork.
+
+    Any commands executed after this function is called will be run in the daemonized process.
+
+    Usage::
+
+        if __name__ == "__main__":
+            daemonize("/tmp/my_daemon.pid")
+            my_func_to_run_daemonized()
+
+    """
+    if pidfile.exists():
+        raise RuntimeError("Already running")
+
+    # First fork (detaches from parent)
+    try:
+        if os.fork() > 0:
+            raise SystemExit(0)  # Parent exit
+    except OSError:
+        raise RuntimeError("fork #1 failed.")
+
+    os.chdir("/")
+    os.umask(0)
+    os.setsid()
+    # Second fork (relinquish session leadership)
+    try:
+        if os.fork() > 0:
+            raise SystemExit(0)
+    except OSError:
+        raise RuntimeError("fork #2 failed.")
+
+    # Flush I/O buffers
+    sys.stdout.flush()
+    sys.stderr.flush()
+
```

Comment:
you should close all file descriptors. see https://stackoverflow.com/questions/17954432/creating-a-daemon-in-linux/17955149#17955149
there's no need for the dup2 calls then